armi/reactor/components/component.py

```diff
@@ -382,13 +367,24 @@ def applyHotHeightDensityReduction(self):
         --------
         self.applyMaterialMassFracsToNumberDensities
         """
-        # this is the same as getThermalExpansionFactor but doesn't fail
-        # on non-fluid materials that have 0 or undefined thermal expansion
-        # (we don't want materials to fail on  __init__ which calls this)
-        axialExpansionFactor = 1.0 + self.material.linearExpansionFactor(
-            self.temperatureInC, self.inputTemperatureInC
-        )
-        self.changeNDensByFactor(1.0 / axialExpansionFactor)
+        if initialColdMaterialExpansion:
+            # material needs to be expanded from the material's cold temp to hot,
+            # not components cold temp, so we don't use mat.linearExpansionFactor or
+            # component.getThermalExpansionFactor.
+            # materials don't typically define the temperature for which their ref dens
+            # is defined so linearExpansionPercent must be called
+            coldMatAxialExpansionFactor = (
+                1.0 + self.material.linearExpansionPercent(Tc=self.temperatureInC) / 100
+            )
+            self.changeNDensByFactor(1.0 / coldMatAxialExpansionFactor)
+        else:
+            # this is the same as getThermalExpansionFactor but doesn't fail
+            # on non-fluid materials that have 0 or undefined thermal expansion
+            # (we don't want materials to fail on  __init__ which calls this)
+            # axialExpansionFactor = 1.0 + self.material.linearExpansionFactor(
+            #     self.temperatureInC, self.inputTemperatureInC
+            # )
+            self.changeNDensByFactor(1.0 / self.getThermalExpansionFactor())
```

Comment:
The `else` seems to only be invoked in tests, as far as I can tell. Could we just remove the `initialColdMaterialExpansion` argument, and then rewrite and/or remove those tests as appropriate?

Reply:
It is invoked in a TP submodule. I would also expect it could also be useful for axial expansion, although it is not used now.

Reply:
would moving the code from the if statement into `applyMaterialMassFracsToNumberDensities` be acceptable, and then the if statement would be eliminated but we would still have a useful hot height density reduction changer.

Reply:
If the code is used then I don't have a problem with it being here. I was just proposing removing the code if in fact it was not being used anymore.